autorest/codegen/serializers/builder_serializer.py

```diff
@@ -929,7 +929,11 @@ def response_headers_and_deserialization(
                 )
             )
         elif response.type:
-            if self.code_model.options["models_mode"]:
+            if self.code_model.options["models_mode"] == "msrest":
+                retval.append(
+                    f"deserialized = self._deserialize('{response.serialization_type}', pipeline_response)"
+                )
+            elif self.code_model.options["models_mode"] == "dpg":
```

Comment:
I don't think we have to go through the deserialization method for DPG models, do you think it would just be better to do `deserialized = DPGModelName(pipeline_response.http_response.json())`

Reply:
Hi @iscai-msft, I'm thinking that `self._deserialize` can help when the response is not a model, for instance when the response is:
- primitive types (string, bool..)
- array of models, or more complex combination of array&dict&model.

We will need to handle these cases here if we don't use the `self._deserialize()`?